internal/system/agent/services.go

```diff
@@ -172,86 +362,90 @@ func getConfig(services []string, ctx context.Context) (ConfigRespMap, error) {
-					clientInfo.Host = e.Host
-					clientInfo.Port = e.Port
-
-					// This code will evolve to take into account a manifest-like functionality in future. So
-					// rather than assume that the runtime bool flag useRegistry has been initialized to true,
-					// given that the flow has reached this point, having already called functions on the Registry,
-					// such as registryClient.IsServiceAvailable(service), we test for its truthiness. I expect
-					// this code to be refactored as we evolve toward a manifest-like functionality in future.
-					usingRegistry := false
-					if registryClient != nil {
-						usingRegistry = true
-					}
-
-					Configuration.Clients[e.ServiceId] = clientInfo
-					params := types.EndpointParams{
-						ServiceKey:  e.ServiceId,
-						Path:        "/",
-						UseRegistry: usingRegistry,
-						Url:         Configuration.Clients[e.ServiceId].Url() + clients.ApiMetricsRoute,
-						Interval:    internal.ClientMonitorDefault,
-					}
-					// TODO: The following note is related to future work:
-					// TODO: With the current deployment strategy, GeneralClient's func init(params types.EndpointParams) {...} returns blank "url"...
-					// TODO: [Need a manifest-like functionality in future...]
-					// Add the service key to the map where the value is the respective GeneralClient
-					generalClients[e.ServiceId] = general.NewGeneralClient(params, startup.Endpoint{RegistryClient: &registryClient})
+				// Preparing to add the specified key to the map where the value will be the respective GeneralClient
+				clientInfo := config.ClientInfo{}
+				clientInfo.Protocol = Configuration.Service.Protocol
+				clientInfo.Host = e.Host
+				clientInfo.Port = e.Port
+
+				// This code will evolve to take into account a manifest-like functionality in future. So
+				// rather than assume that the runtime bool flag useRegistry has been initialized to true,
+				// given that the flow has reached this point, having already called functions on the Registry,
+				// such as registryClient.IsServiceAvailable(service), we test for its truthiness. I expect
+				// this code to be refactored as we evolve toward a manifest-like functionality in future.
+				usingRegistry := false
+				if registryClient != nil {
+					usingRegistry = true
+				}
 
-					responseJSON, err := generalClients[e.ServiceId].FetchMetrics(ctx)
-					if err != nil {
-						m.Metrics[service] = fmt.Sprintf(err.Error())
-						LoggingClient.Error(err.Error())
-					} else {
-						m.Metrics[service] = ProcessResponse(responseJSON)
-					}
+				Configuration.Clients[e.ServiceId] = clientInfo
+				params := types.EndpointParams{
+					ServiceKey:  e.ServiceId,
+					Path:        "/",
+					UseRegistry: usingRegistry,
+					Url:         Configuration.Clients[e.ServiceId].Url() + clients.ApiMetricsRoute,
+					Interval:    internal.ClientMonitorDefault,
 				}
+				// TODO: The following note is related to future work:
```

Comment:
I'd suggest refactoring these comment to just be a single line comment TODO, I think it's clear by the TODO that it's future work, and it's not clear what is meant by manifest-like functionality. In addition to those points, there's already a comment above about future manifest work so this seems duplicative.